iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -1273,6 +1274,510 @@ public void testScanTableCaseInsensitive() throws IOException {
+
+    // Run a 'select *' to check if the order of the column in the result has been changed.
+    List<Object[]> rows = shell.executeStatement("SELECT * FROM default.customers");
+    HiveIcebergTestUtils.validateData(customersWithLastNameFirst,
+        HiveIcebergTestUtils.valueForRow(customerSchemaLastNameFirst, rows), 1);
+
+    // Query the data with names and check if the result is the same as when the table was created.
+    rows = shell.executeStatement("SELECT customer_id, first_name, last_name FROM default.customers");
+    HiveIcebergTestUtils.validateData(HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS,
+        HiveIcebergTestUtils.valueForRow(HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA, rows), 0);
+
+    // Insert data from Hive to check if the last_name column has to be before the customer_id in the values list.
+    shell.executeStatement("INSERT INTO default.customers values ('Lily', 3L, 'Magenta')");
+
+    customersWithLastNameFirstBuilder.add("Lily", 3L, "Magenta");
+    customersWithLastNameFirst = customersWithLastNameFirstBuilder.build();
+    rows = shell.executeStatement("SELECT * FROM default.customers");
+    HiveIcebergTestUtils.validateData(customersWithLastNameFirst,
+        HiveIcebergTestUtils.valueForRow(customerSchemaLastNameFirst, rows), 1);
+  }
+
+  @Test
+  public void testUpdateColumnTypeInIcebergTable() throws IOException {
+    // Create an Iceberg table with int, float and decimal(2,1) types with some initial records
+    Schema schema = new Schema(optional(1, "id", Types.LongType.get()),
+        optional(2, "int_col", Types.IntegerType.get(), "This is an integer type"),
+        optional(3, "float_col", Types.FloatType.get(), "This is a float type"),
+        optional(4, "decimal_col", Types.DecimalType.of(2, 1), "This is a decimal type"));
+
+    List<Record> records = TestHelper.RecordsBuilder.newInstance(schema).add(0L, 35, 22F, BigDecimal.valueOf(13L, 1))
+        .add(1L, 223344, 555.22F, BigDecimal.valueOf(22L, 1)).add(2L, -234, -342F, BigDecimal.valueOf(-12L, 1)).build();
+
+    Table icebergTable = testTables.createTable(shell, "types_table", schema, fileFormat, records);
+
+    Schema schemaForResultSet =
+        new Schema(optional(1, "id", Types.LongType.get()), optional(2, "int_col", Types.IntegerType.get()),
+            optional(3, "float_col", Types.DoubleType.get()), optional(4, "decimal_col", Types.StringType.get()));
+
+    List<Record> expectedResults = TestHelper.RecordsBuilder.newInstance(schema).add(0L, 35, 22d, "1.3")
+        .add(1L, 223344, 555.22d, "2.2").add(2L, -234, -342d, "-1.2").build();
+
+    // Check the select resut and the column types from Hive
+    List<Object[]> rows = shell.executeStatement("SELECT * FROM types_table");
+    HiveIcebergTestUtils.validateData(expectedResults, HiveIcebergTestUtils.valueForRow(schemaForResultSet, rows), 0);
+
+    rows = shell.executeStatement("DESCRIBE types_table");
+    Assert.assertEquals("id", rows.get(0)[0]);
```

Comment:
you might want to use `shell.metastore().getTable()` to load the HMS table and then access its columns directly `table.getSd().getCols()`. But it's up to you. Just generally, we've avoided parsing the describe output so far

Reply:
Oh, ok, thanks for the hint. I will change that check.